sale_block_no_stock/models/sale_order.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import _, exceptions, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        qty_exceptions = []
+        for record in self:
+            if any(
+                self.env.user in group.users
+                for group in record.company_id.sale_line_block_allowed_groups
+            ):
```

Comment:
suggestion: perf improvement

```suggestion
            if self.env.user in record.company_id.sale_line_block_allowed_groups.users:
```

Reply:
Doesn't work like this. I've tryied this solution before